tests/server/cache_test.py

```diff
@@ -13,30 +13,104 @@
 class TestCache:
     """Tests the server cache."""
 
-    def test(self):
+    def test_simple(self):
         with TemporaryDirectory() as tmp_dir:
-            cache_dir = Path(tmp_dir)
-            cache = Cache(cache_dir)
+            root_dir = Path(tmp_dir)
+            cache = Cache(root_dir, 1000)
+            cache_dir = root_dir / "cache"
 
             file1 = bytearray([0x1, 0x2, 0x3, 0x9])
             cache.put("hash1", file1)
 
-            assert cache.get("hash1") == str(cache_dir / "cache" / "hash1")
+            assert cache.get("hash1") == str(cache_dir / "hash1")
             assert "hash1" in cache
             assert Path.read_bytes(Path(cache.get("hash1"))) == file1
 
             file2 = bytearray([0x3, 0x6, 0x3, 0x9])
             cache.put("hash2", file2)
 
-            assert cache.get("hash2") == str(cache_dir / "cache" / "hash2")
+            assert cache.get("hash2") == str(cache_dir / "hash2")
             assert "hash2" in cache
             assert Path.read_bytes(Path(cache.get("hash2"))) == file2
 
             file3 = bytearray([0x4, 0x2])
             cache.put("hash3", file3)
 
-            assert cache.get("hash3") == str(cache_dir / "cache" / "hash3")
+            assert cache.get("hash3") == str(cache_dir / "hash3")
             assert "hash3" in cache
             assert Path.read_bytes(Path(cache.get("hash3"))) == file3
 
             assert "other_hash" not in cache
+
+    @staticmethod
+    def assert_hash_in_cache(cache: Cache, hash_value: str):
+        assert hash_value in cache
+        assert (cache.cache_folder / hash_value).exists()
+
+    @staticmethod
+    def assert_hash_not_in_cache(cache: Cache, hash_value: str):
+        assert hash_value not in cache
+        assert not (cache.cache_folder / hash_value).exists()
+
+    def test_eviction_size_limit(self):
+        with TemporaryDirectory() as tmp_dir:
```

Comment:
Maybe could use the `pytest` `tmpdir` fixture instead: https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture